tests/rsapss.cc

```diff
@@ -0,0 +1,214 @@
+#include "Hacl_RSAPSS.h"
+#include "Hacl_Spec.h"
+#include "util.h"
+
+using json = nlohmann::json;
+
+typedef struct
+{
+  uint32_t tcId;
+  bytes msg;
+  bytes sgnt;
+  bool expected;
+} Test;
+
+typedef struct
+{
+  bytes e;
+  bytes n;
+  uint32_t saltLen;
+  Spec_Hash_Definitions_hash_alg a;
+  std::vector<Test> tests;
+} Group;
+
+void
+verify(bytes e,
+       bytes n,
+       uint32_t saltLen,
+       Spec_Hash_Definitions_hash_alg alg,
+       bytes msg,
+       bytes sgnt,
+       bool* out)
+{
+  uint64_t* pkey = Hacl_RSAPSS_new_rsapss_load_pkey(
+    n.size() * 8, e.size() * 8, n.data(), e.data());
+
+  bool got1 = Hacl_RSAPSS_rsapss_verify(alg,
+                                        n.size() * 8,
+                                        e.size() * 8,
+                                        pkey,
+                                        saltLen,
+                                        sgnt.size(),
+                                        sgnt.data(),
+                                        msg.size(),
+                                        msg.data());
+
+  bool got2 = Hacl_RSAPSS_rsapss_pkey_verify(alg,
+                                             n.size() * 8,
+                                             e.size() * 8,
+                                             n.data(),
+                                             e.data(),
+                                             saltLen,
+                                             sgnt.size(),
+                                             sgnt.data(),
+                                             msg.size(),
+                                             msg.data());
+
+  ASSERT_EQ(got1, got2) << "`Hacl_RSAPSS_rsapss_verify(...)` deviates from "
+                           "`Hacl_RSAPSS_rsapss_pkey_verify(...)`.";
+
+  *out = got2;
+}
+
+class RsaPssVerifySuite : public ::testing::TestWithParam<Group>
+{};
+
+TEST_P(RsaPssVerifySuite, Group)
+{
+  auto group = GetParam();
+
+  // std::cout << "e: " << bytes_to_hex(group.e) << std::endl
+  //           << "n: " << bytes_to_hex(group.n) << std::endl;
+
+  for (auto test : group.tests) {
+    // std::cout << "msg: " << bytes_to_hex(test.msg) << std::endl
+    //           << "sgnt: " << bytes_to_hex(test.sgnt) << std::endl;
+
+    bool got;
+    verify(group.e, group.n, group.saltLen, group.a, test.msg, test.sgnt, &got);
+
+    EXPECT_EQ(test.expected, got) << "tcId=" << test.tcId;
+  }
+}
+
+std::vector<Group>
+read_json(char* path)
+{
+  json tests;
+  std::ifstream file(path);
+  file >> tests;
+
+  std::vector<Group> testGroups;
+
+  for (auto& group_raw : tests["testGroups"].items()) {
+    auto group = group_raw.value();
+
+    bytes e = from_hex(group["e"]);
+    bytes n = from_hex(group["n"]);
+
+    // Remove first 0x00 byte in n.
+    if (n[0] == 0x00) {
+      n.erase(n.begin());
+    } else {
+      std::ostringstream msg;
+      msg << "Expected first byte of \"n\" to be 0x00 in Wycheproof test (path=" << path << ").";
```

Comment:
clang format breaks the line here.

Reply:
clang-format is now happy on my machine. Can you recheck?